Bottom: 724450b1f64d5cdc293b8ecabf3dc1c7c8fae099
Top:    68e4f81ee670b5734383b05884f10b7e49a8de75
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 17:25:26 +0000

Prepend when adding things to lists


---

diff --git a/src/hid/step/object3d_step.c b/src/hid/step/object3d_step.c
index d47a8b4..ee8938b 100644
--- a/src/hid/step/object3d_step.c
+++ b/src/hid/step/object3d_step.c
@@ -295,10 +295,11 @@ object3d_to_step_body_fragment (step_file *step,
           edge = contour->first_edge;
           do
             {
-              edge_loop_edges = g_list_append (edge_loop_edges, GINT_TO_POINTER (ORIENTED_EDGE_IDENTIFIER (edge)));
+              edge_loop_edges = g_list_prepend (edge_loop_edges, GINT_TO_POINTER (ORIENTED_EDGE_IDENTIFIER (edge)));
             }
           while (edge = LNEXT (edge), edge != contour->first_edge);
 
+          edge_loop_edges = g_list_reverse (edge_loop_edges);
           edge_loop = step_edge_loop (step, "NONE", edge_loop_edges);
 
           if (outer_contour)
@@ -306,13 +307,16 @@ object3d_to_step_body_fragment (step_file *step,
           else
             contour->face_bound_identifier = step_face_bound (step, "NONE", edge_loop, true);
 
-          face_contour_list = g_list_append (face_contour_list, GINT_TO_POINTER (contour->face_bound_identifier));
+          face_contour_list = g_list_prepend (face_contour_list, GINT_TO_POINTER (contour->face_bound_identifier));
         }
 
+      face_contour_list = g_list_reverse (face_contour_list);
       face->face_identifier = step_advanced_face (step, "NONE", face_contour_list, face->surface_identifier, !face->surface_orientation_reversed);
-      shell_face_list = g_list_append (shell_face_list, GINT_TO_POINTER (face->face_identifier));
+      shell_face_list = g_list_prepend (shell_face_list, GINT_TO_POINTER (face->face_identifier));
     }
 
+  shell_face_list = g_list_reverse (shell_face_list);
+
   /* Closed shell which bounds the brep solid */
   pcb_shell_identifier = step_closed_shell (step, "NONE", shell_face_list);
   brep_identifier = step_manifold_solid_brep (step, body_name /* This is picked up as the solid body name by Solidworks */, pcb_shell_identifier);
